src/bpfprogram.cpp

```diff
@@ -290,4 +286,280 @@ void BpfProgram::appendFileFuncInfos(const struct btf_ext_info_sec *src,
+            LOG(WARNING) << msg << ", skipping.";
+            return;
+          } else
+            // explicit match failed, fail hard
+            throw FatalUserException(msg);
+        }
+
+        opts.attach_btf_id = btf_id.first;
+        opts.attach_btf_obj_fd = btf_id.second;
+      } else {
+        opts.kern_version = kernel_version(load_config.kver);
+      }
+      logbuf << "load " << probe.name << (load_config.btf ? ", with BTF" : "")
+             << (load_config.func_infos ? ", with func_infos" : "");
+      if (opts.kern_version) {
+        logbuf << ", version: " << ((opts.kern_version >> 16) & 0xFF) << "."
+               << ((opts.kern_version >> 8) & 0xFF) << "."
+               << (opts.kern_version & 0xFF);
+      }
+
+      {
+        // Redirect stderr, so we don't get error messages from libbpf
+        StderrSilencer silencer;
+        if (bt_debug == DebugLevel::kNone)
+          silencer.silence();
+
+        int btf_fd = -1;
+        if (load_config.btf) {
+          LIBBPF_OPTS(bpf_btf_load_opts,
+                      btf_opts,
+                      .log_buf = log_buf.get(),
+                      .log_level = static_cast<__u32>(log_level),
+                      .log_size = static_cast<__u32>(log_buf_size), );
+
+          auto &btf_sec = bytecode.getSection(".BTF");
+          btf_fd = bpf_btf_load(btf_sec.data(), btf_sec.size(), &btf_opts);
+          if (btf_fd < 0) {
+            logbuf << ": BTF load failed";
+            goto log_error; // and continue to the next load_config fallback
+          }
+          opts.prog_btf_fd = (__u32)btf_fd;
+        }
+
+        // Skip .BTF.ext func_info relocation on fallback for older kernels
+        if (load_config.func_infos && !func_infos.empty()) {
+          opts.func_info_rec_size = sizeof(struct bpf_func_info);
+          opts.func_info = func_infos.data();
+          opts.func_info_cnt = func_infos.size() / sizeof(struct bpf_func_info);
+        }
+
+        // Load the BPF program on BTF load success or on fallback without it
+        if (!load_config.btf || btf_fd >= 0) {
+          fd_ = bpf_prog_load(static_cast<::bpf_prog_type>(prog_type),
+                              name.c_str(),
+                              license,
+                              reinterpret_cast<const struct bpf_insn *>(
+                                  insns.data()),
+                              insns.size() / sizeof(struct bpf_insn),
+                              &opts);
+          if (btf_fd >= 0)
+            close(btf_fd);
+        }
+      }
+
+      if (opts.attach_btf_obj_fd > 0)
+        close(opts.attach_btf_obj_fd);
+      if (fd_ >= 0)
+        break;
+      logbuf << " = " << fd_ << ": " << std::strerror(-fd_);
+    log_error:
+      if (!logbuf.str().empty()) { // log the error of the failed attempt
+        LOG(V1) << logbuf.str();
+        logbuf.str("");
+      }
+    } // obsolete indendation level, has no effect, kept for a small commitdiff
```

Comment:
I don't think we need to keep this any more

Reply:
Will also go away soon but sure, let's drop it.